lib/buffer.js

```diff
@@ -335,6 +335,14 @@ Buffer.byteLength = byteLength;
 
 // toString(encoding, start=0, end=buffer.length)
 Buffer.prototype.toString = function(encoding, start, end) {
+  // .toString() can be called by JS on prototype (Buffer.prototype + '').
+  if (!(this instanceof Buffer)) {
+    if (arguments.length === 0)
```

Comment:
I still don't understand the purpose of this check at all. Why not always return O.p.toString when used on a non-buffer.